bundles/specmate-migration-test/src/com/specmate/migration/test/support/TestMigratorImpl.java

```diff
@@ -60,108 +63,113 @@ public void migrate(Connection connection) throws SpecmateException {
 		aAdded.migrateNewDoubleAttribute("diagram", "length", null);
 		aAdded.migrateNewIntegerAttribute("diagram", "amount", -1);
 		aAdded.migrateNewIntegerAttribute("diagram", "intamount", -1);
 		aAdded.migrateNewDoubleAttribute("diagram", "doublelength", 0.0);
 		aAdded.migrateNewBooleanAttribute("diagram", "booleanlinked", false);
 	}
-	
+
 	private void migrateObjectAdded(Connection connection) throws SpecmateException {
 		String objectName = "Document";
-		ObjectToSQLMapper oAdded = new ObjectToSQLMapper(connection, logService, packageName, getSourceVersion(), 
+		IObjectToSQLMapper oAdded = dbProvider.getObjectToSQLMapper(packageName, getSourceVersion(),
 				getTargetVersion());
 		oAdded.newObject(objectName);
-		
-		AttributeToSQLMapper aAdded = new AttributeToSQLMapper(connection, logService, packageName, getSourceVersion(), 
+
+		IAttributeToSQLMapper aAdded = dbProvider.getAttributeToSQLMapper(packageName, getSourceVersion(),
 				getTargetVersion());
 		aAdded.migrateNewStringAttribute(objectName, "id", "");
 		aAdded.migrateNewBooleanAttribute(objectName, "tested", false);
 		aAdded.migrateNewLongAttribute(objectName, "length", null);
 		aAdded.migrateNewStringAttribute(objectName, "owner", null);
 		aAdded.migrateNewReference(objectName, "contents");
 	}
-	
+
 	private void migrateAttributeRenamed(Connection connection) throws SpecmateException {
-		AttributeToSQLMapper aRenamed = new AttributeToSQLMapper(connection, logService, packageName, 
-				getSourceVersion(), getTargetVersion());
+		IAttributeToSQLMapper aRenamed = dbProvider.getAttributeToSQLMapper(packageName, getSourceVersion(),
+				getTargetVersion());
 		aRenamed.migrateRenameAttribute("Diagram", "tested", "istested");
-		aRenamed.migrateRenameAttribute("File", "tested", "istested");
+		aRenamed.migrateRenameAttribute("Sketch", "tested", "istested");
 	}
-	
+
 	private void migrateTypesChanged(Connection connection) throws SpecmateException {
-		AttributeToSQLMapper aTypeChanged = new AttributeToSQLMapper(connection, logService, packageName, 
-				getSourceVersion(), getTargetVersion());
-				
-		aTypeChanged.migrateChangeType("File", "shortVar1", EDataType.INT);
-		aTypeChanged.migrateChangeType("File", "shortVar2", EDataType.LONG);
-		aTypeChanged.migrateChangeType("File", "shortVar3", EDataType.FLOAT);
-		aTypeChanged.migrateChangeType("File", "shortVar4", EDataType.DOUBLE);
-		
-		aTypeChanged.migrateChangeType("File", "charVar1", EDataType.INT);
-		aTypeChanged.migrateChangeType("File", "charVar2", EDataType.LONG);
-		aTypeChanged.migrateChangeType("File", "charVar3", EDataType.FLOAT);
-		aTypeChanged.migrateChangeType("File", "charVar4", EDataType.DOUBLE);
-		EDataType charVar5 = EDataType.STRING;
+		IAttributeToSQLMapper aTypeChanged = dbProvider.getAttributeToSQLMapper(packageName, getSourceVersion(),
+				getTargetVersion());
+
+		aTypeChanged.migrateChangeType("Sketch", "shortVar1", H2DataType.INT);
```

Comment:
You could extract "Sketch" into a constant or use ArtefactPackage.Literals.Sketch.getName()

Reply:
Fixed.